logstash-core/src/main/java/org/logstash/common/io/ByteArrayStreamOutput.java

```diff
@@ -0,0 +1,65 @@
+package org.logstash.common.io;
+
+import java.nio.ByteBuffer;
+
+public class ByteArrayStreamOutput extends StreamOutput {
+    private byte[] bytes;
+
+    private int pos;
```

Comment:
actually, only MemoryPageIOStream is unused. all the rest of the *Stream classes are used for FileCheckpointIO
@guyboertje please followup